runtime/src/common/sgx/seal.rs

```diff
@@ -49,11 +50,16 @@ pub fn unseal(key_policy: Keypolicy, context: &[u8], ciphertext: &[u8]) -> Optio
     let ciphertext = &ciphertext[..ct_len];
 
     let d2 = new_deoxysii(key_policy, context);
-    let plaintext = d2
-        .open(&nonce, ciphertext.to_vec(), vec![])
-        .expect("ciphertext is corrupted");
 
-    Some(plaintext)
+    match d2.open(&nonce, ciphertext.to_vec(), vec![]) {
+        Ok(plaintext) => Ok(Some(plaintext)),
+        Err(_) => {
+            let err_msg = "ciphertext is corrupted (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and resync the node)";
```

Comment:
This works.
```
{"level":"error","module":"runtime/sgx","msg":"ciphertext is corrupted (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and resync the node)","runtime_id":"c000000000000000ffffffffffffffffffffffffffffffffffffffffffffffff","runtime_name":"test-runtime","ts":"2023-12-12T09:48:56.7323321Z"}
```
However, I would maybe try to avoid logging the message here and log it in the verifier, when we try to load trusted state. The logger is already available there, and this should still give use desired result, as in production, all confidential runtimes use trust roots and try to load them on startup.

Reply:
Also instead of "and resync the node" it should be "and restart the node" which is more clear IMO.

Reply:
To clarify, my comment suggested to move logging to method [run](https://github.com/oasisprotocol/oasis-core/blob/master/runtime/src/consensus/tendermint/verifier/mod.rs#L651 ) if `load` fails (e.g. wrap error message with `trusted root failed (if running in SGX mode ...)`) while keeping the method `load` intact.

Reply:
Done, please retry if it works under SGX :)

Reply:
Tested on SGX by simulating CPU change. Simulation changed deoxys key on node restart, so decryption of encrypted data failed.

Compute nodes need encryption only for trusted state.
```
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"8000000000000000000000000000000000000000000000000000000000000000","runtime_name":"test-runtime","ts":"2024-01-09T11:43:55.528983586Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"8000000000000000000000000000000000000000000000000000000000000000","runtime_name":"test-runtime","ts":"2024-01-09T11:43:56.47622053Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"8000000000000000000000000000000000000000000000000000000000000000","runtime_name":"test-runtime","ts":"2024-01-09T11:43:57.603294562Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"8000000000000000000000000000000000000000000000000000000000000000","runtime_name":"test-runtime","ts":"2024-01-09T11:43:59.719795683Z"}
...
```
Key managers need encryption for trusted state, policy and master secret proposal. As trusted state is decrypted first, the solution works.
```
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"c000000000000000ffffffffffffffffffffffffffffffffffffffffffffffff","runtime_name":"test-runtime","ts":"2024-01-09T11:56:22.696531759Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"c000000000000000ffffffffffffffffffffffffffffffffffffffffffffffff","runtime_name":"test-runtime","ts":"2024-01-09T11:56:23.716499547Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"c000000000000000ffffffffffffffffffffffffffffffffffffffffffffffff","runtime_name":"test-runtime","ts":"2024-01-09T11:56:24.693976049Z"}
{"level":"error","module":"runtime/consensus/cometbft/verifier","msg":"failed to load trusted state (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and restart the node)","runtime_id":"c000000000000000ffffffffffffffffffffffffffffffffffffffffffffffff","runtime_name":"test-runtime","ts":"2024-01-09T11:56:26.624439067Z"}
...
```